include/libdnf5/base/base.hpp

```diff
@@ -134,6 +135,20 @@ class LIBDNF_API Base {
 
     libdnf5::BaseWeakPtr get_weak_ptr();
 
+    /// @brief Load libdnf5 plugin config, extract name of the plugin and check if it is enabled
+    /// @param config_file_path Path to a plugin config
+    /// @return a tuple with plugin name, parsed config and a bool whether the plugin is enabled
+    std::tuple<std::string, libdnf5::ConfigParser, bool> load_plugin_config(const std::string & config_file_path);
+
+    /// @brief Add libdnf5 plugin instance that introduces additional logic into the library using hooks.
+    /// @param plugin_name Name of the new plugin
+    /// @param plugin_config Parsed config of the new plugin
+    /// @param iplugin_instance New libdnf5 plugin instance
+    void add_plugin(
```

Comment:
I suggest changing the API to this: `void add_plugin(std::string plugin_name, libdnf5::ConfigParser && plugin_config, libdnf5::plugin::IPlugin & iplugin_instance);`

 Specifically, pass plugin_config using `&&`. Inside the method, a `move` is performed. The call in `python_plugins_loader` will also need to be updated to work with the revised API and to handle memory correctly.